tests/installation/usb_install.pm

```diff
@@ -0,0 +1,112 @@
+    my $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/ignition/config.ign")
+      . " -o /mnt/ignition/config.ign";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    assert_script_run("ls /mnt/ignition/config.ign");
+    save_screenshot;
+
+    # Write combustion script to the same usb with ignition
+    assert_script_run("mkdir -p /mnt/combustion");
+    $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/combustion/script")
+      . " -o /mnt/combustion/script";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    my $SERIALCONSOLE = get_required_var('SERIALCONSOLE');
+    assert_script_run("sed -i 's/SERIALCONSOLE/$SERIALCONSOLE/g' /mnt/combustion/script");
+    assert_script_run("chmod a+x /mnt/combustion/script");
+    assert_script_run("ls -l /mnt/combustion/script");
+    save_screenshot;
+    record_info('Ignition and combustion content:', script_output('cat /mnt/ignition/config.ign /mnt/combustion/script'));
+
+    # Flush to usb stick
+    assert_script_run("sync");
+    assert_script_run("umount -l /mnt");
+    record_info("Ignition and combustion files are successfully downloaded and written to usb $provision_usb.");
+
+    # Download and dd the iso to the other usb
+    my $download_url = "http://" . get_var('OPENQA_URL', get_var('OPENQA_HOSTNAME')) . "/assets/iso/" . get_required_var('ISO');
+    die "ISO URL is not accessible: $download_url." unless head($download_url);
+    $cmd = "curl -L $download_url | dd of=$medium_usb bs=1M";
+    script_retry($cmd, retry => 2, delay => 10, timeout => 600, die => 1);
+    save_screenshot;
+
+    # Flush
+    assert_script_run("sync");
+    record_info("ISO is saved successfully to usb $medium_usb.");
+
+
+    # Set next boot to usb
+    if (check_var('IPXE_UEFI', '1')) {
+        # Some machines do not support setting floppy boot via ipmitool in uefi boot mode,
+        # so we use efibootmgr instead
+        record_info('efibootmgr output after dd iso to usb:', script_output('efibootmgr'));
+        save_screenshot;
+        # Some machines are problematic which show faulty boot entries as active boot.
+        # Add workaround USB_BOOT_DEV_KEYWORD to filter correct boot.
+        my $cmd = 'efibootmgr | grep usb -i | grep -v generic -i | grep "\*"';
+        $cmd .= get_var('USB_BOOT_DEV_KEYWORD') ? ' | grep ' . get_var('USB_BOOT_DEV_KEYWORD') : '';
+        my $usb_boot = script_output("$cmd");
+        save_screenshot;
+        die "Only 1 bootable USB should be here. But we find in efibootmgr output: $usb_boot." if (!$usb_boot || $usb_boot =~ /\n/);
+        $usb_boot =~ /Boot([0-9A-F]+)\*/m;
+        my $usb_boot_num = $1;
+        assert_script_run("efibootmgr -n $usb_boot_num");
+        save_screenshot;
+        record_info('efibootmgr output after setting next boot to usb:', script_output('efibootmgr'));
```

Comment:
The consistent way to use usb boot is:
1. lsblk with "-o RM“ option to list removable disk, namely usb in your case.
2. Then add a new efi boot entry using efibootmgr, for example:
efibootmgr -c -d /dev/USBDISK -p 1 -L "LABEL" -l "BOOTLOADER“
3. Boot from above newly added boot entry.

Reply:
The efi boot entry is already automatically added by efibootmgr. But it is not easy to tell which is which . I tried to find a method, but not easy from pci or usb numbers.

This is vh015 output, Rear USB 1 ( /dev/sdb) is the usb with boot iso.


```
vh015:~ # ls /dev/disk/by-path -l
total 0
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:03:00.3-usb-0:1.1:1.0-scsi-0:0:0:0 -> ../../sdb
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:03:00.3-usb-0:1.1:1.0-scsi-0:0:0:0-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:03:00.3-usb-0:1.1:1.0-scsi-0:0:0:0-part2 -> ../../sdb2
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:03:00.3-usb-0:1.3:1.0-scsi-0:0:0:0 -> ../../sdd
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:03:00.3-usb-0:2.1:1.0-scsi-0:0:0:0 -> ../../sdc
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:03:00.3-usbv3-0:1.1:1.0-scsi-0:0:0:0 -> ../../sdb
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:03:00.3-usbv3-0:1.1:1.0-scsi-0:0:0:0-part1 -> ../../sdb1
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:03:00.3-usbv3-0:1.1:1.0-scsi-0:0:0:0-part2 -> ../../sdb2
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:03:00.3-usbv3-0:1.3:1.0-scsi-0:0:0:0 -> ../../sdd
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:03:00.3-usbv3-0:2.1:1.0-scsi-0:0:0:0 -> ../../sdc
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 pci-0000:23:00.0-sas-0x50000f0a47804242-lun-0 -> ../../sda
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:23:00.0-sas-0x50000f0a47804242-lun-0-part1 -> ../../sda1
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:23:00.0-sas-0x50000f0a47804242-lun-0-part2 -> ../../sda2
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 pci-0000:23:00.0-sas-0x50000f0a47804242-lun-0-part3 -> ../../sda3
vh015:~ # ls /dev/disk/by-uuid
2024-02-20-15-33-22-00  2d726d5f-c35f-42cb-a9f7-58d40f92401a  77B7-1ADB  8B62-3D11  eb93edf6-0168-4db8-8e03-f170da8e6f76
vh015:~ # ls /dev/disk/by-uuid -l
total 0
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 2024-02-20-15-33-22-00 -> ../../sdb1
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 2d726d5f-c35f-42cb-a9f7-58d40f92401a -> ../../sda3
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 77B7-1ADB -> ../../sda2
lrwxrwxrwx. 1 root root 10 Mar 27 07:19 8B62-3D11 -> ../../sdb2
lrwxrwxrwx. 1 root root  9 Mar 27 07:19 eb93edf6-0168-4db8-8e03-f170da8e6f76 -> ../../sdd

vh015:~ # efibootmgr -v |grep usb -i
Boot000A* Generic USB Boot      UsbClass(ffff,ffff,255,255)
Boot000B* Rear USB 1 : Kingston DataTraveler 3.0        PciRoot(0x0)/Pci(0x7,0x1)/Pci(0x0,0x3)/USB(2,0)/USB(0,0){auto_created_boot_option}
Boot000C* Front USB 1 : Kingston DataTraveler 3.0       PciRoot(0x0)/Pci(0x7,0x1)/Pci(0x0,0x3)/USB(2,0)/USB(2,0){auto_created_boot_option}
Boot0016* Internal SD Card 1 : Generic USB3.0-CRW       PciRoot(0x0)/Pci(0x7,0x1)/Pci(0x0,0x3)/USB(3,0)/USB(0,0){auto_created_boot_option}
vh015:~ # 
```

Reply:
I never thought about the validness of adding a new boot entry when there was already an automatically added one. But why not? It is indeed much easier to tell which is the usb boot.
Code is done. Please check it out. New verification run : http://10.67.129.77/tests/325.